pkg/plugins/user.go

```diff
@@ -101,41 +102,52 @@ func createUser(fs vfs.FS, u schema.User, console Console) error {
 		}
 	}
 
+	if u.Homedir == "" {
+		u.Homedir = fmt.Sprintf("%s/%s", usrDefaults["HOME"], u.Name)
+	}
+
 	uid := -1
+
+	// If UID is specified just put it there. No matter whats in the system or the collisions. Good luck.
 	if u.UID != "" {
 		// User defined-uid
 		uid, err = strconv.Atoi(u.UID)
 		if err != nil {
 			return errors.Wrap(err, "invalid uid defined")
 		}
 	} else {
+		// Try to get the existing UID in the system
 		list := users.NewUserList()
 		list.SetPath(etcpasswd)
 		list.Load()
-
 		user := list.Get(u.Name)
-
 		if user != nil {
 			uid, err = user.UID()
 			if err != nil {
 				return errors.Wrap(err, "could not get user id")
 			}
 		} else {
-			// https://systemd.io/UIDS-GIDS/#special-distribution-uid-ranges
-			uid, err = list.GenerateUIDInRange(entities.HumanIDMin, entities.HumanIDMax)
-			if err != nil {
-				return errors.Wrap(err, "no available uid")
+			// Try to see if the user was created previously with a given UID by checking for an existing home dir
```

Comment:
won't this be problematic since you can define a user like:

```
stages:
   default:
     - name: "Setup users"
       users: 
          bastion: 
            passwd: "strongpassword"
            homedir: "/home/foo
  ```

Reply:
everything is problematic in here, thats what happens when you give users too much choice. They dont know whats better for them.

We are just trying harder to get it if its there, unfortunately we cannot cover all angles. What happens if during one boot the user has the default dir and the next one he modified to now set the homedir specifically in the config? All their files are gone? On a different dir? What if he also changes the uid? Now they are gone and also he cant access them anymore?

Too many different paths and scenarios due to the broad config spectrum of the user.